### DISCUSSION TITLE
How to get all data stored in a collection?

### DISCUSSION THREAD

**Discussion by kumsumit** [isar/isar#820] [Help] [answered]
I am using Isar 3.0.2. 
Tried  this, but didn't works  

```
final firstTenShoes = await isar.shoes.where().findAll();
```

Tried this, but this doesn't work too.
```
  final firstTenShoes = await isar.shoes.buildQuery<Shoe>().findAll());
```

**Comment by Djihanegh** [comment, marked as answer]
try this : 

` await isar.txn(() async {
          firstTenShoes = await isar.shoes.where().findAll();
        });`

**Reply by kumsumit** [reply]
The issue is that , I am getting error findAll is not defined for QueryBuilder Type , and the type of where is QueryBuildType

**Reply by kumsumit** [reply]
So, I can't use findAll with where(), I tried to use with collection but didn't succeed there too. 
`isar.shoes.findAll()` didn't work too.

**Reply by Djihanegh** [reply]
i think the type of firstTenShoes is missing ; 

here's an example : 

``` dart static Future<List<Apps>> getAllApps() async {
    final isar = await db;
    List<Apps> apps = [];
    if (isar != null) {
      try {
        await isar.txn(() async {
          apps = await isar.apps.where().findAll();
        });
      } catch (e) {
        log(e.toString());
      }
    }
    return apps;
  }

**Reply by kumsumit** [reply]
Ah, I am unfortunate enough that I am not even getting suggestion of `findAll` after `where()` or `collection()`

**Reply by kumsumit** [reply]
@leisim , Can you please confirm that Isar documentation is compatible with latest version of Isar.
Because, Earlier, when I was using 2.x.y version ,  `await isar.shoes.findAll()` was working fine and I was getting suggestion of findAll(), But When ( Currently ) I am using 3.0.2 it niether giving me suggestion of findAll() not running.

**Reply by simc** [reply]
You are missing some imports. Make sure to import the model class and Isar. The docs are up-to-date and findAll is working. Countless unit tests verify it 🙂

**Reply by kumsumit** [reply]
@leisim , @Djihanegh , you were true. I was missing import  for Isar.

**Reply by kumsumit** [reply]
Thanks .

**Comment by anmol-me** [comment]
With version isar 3.1.0+1 use,


  ```
Future<List<TodoModel>> fetchTodos() async {
    List<TodoModel> todos = [];

    await isar.writeTxn(() async {
      todos = await isar.todoModels.where().findAll();
    });

    return todos;
  }

```
